tests/test_log.py

```diff
@@ -160,3 +160,54 @@ def test_log_grep_bad_path(mock_repo):
     with patch('git.repo.fun.name_to_object'):
         with pytest.raises(exceptions.FileDoesntExistException):
             repo.log.grep_for_commits("test_branch", 'test', True, 'wrongpath')
+
+
+def test_log_show_commit(mock_repo, fake_commits):
+    """
+    GIVEN GitRepo is initialized with a path and repo
+    WHEN log.log_show_commit is called with correct parameters
+    THEN a single string should be returned
+    """
+    mock_repo.commit.return_value = fake_commits[1]
+    repo = GitRepo(repo=mock_repo)
+
+    with patch('git.repo.fun.name_to_object'):
+        change = repo.log.log_show_commit('0010000000000000000')
+
+    assert change == (
+        "commit 0010000000000000000\n"
+        "Author: Test Author <testauthor@example.com>\n"
+        "Date: Wed Dec 05 10:36:19 2018 \n\n"
+        "This is a commit message (#1)\nWith some details."
+    )
+
+
+def test_log_show_commit_no_commit(mock_repo):
+    """
+    GIVEN GitRepo is initialized with a path and repo
+    WHEN log.log_show_commit is called with an unknown commit hash
+    THEN an empty list is returned
```

Comment:
This should be "empty string" after the changes, though as per my comment above I wonder if the mock shouldn't raise an exception instead now?

Reply:
hm... in fact... Due to the way I mock this: ```mock_repo.commit.return_value = ''``` it won't raise anything.... Maybe that's only possible in the integration test?